### ISSUE TITLE
Change of behavior between jarchivelib and compress4j

### ISSUE THREAD

**Issue by hhslepicka** [compress4j/compress4j#18]
Hi,

Thank you for taking on the development of this library and supporting it.

I recently switched from jarchivelib to compress4j due to the vulnerability warnings for one of my projects. After doing that, I noticed that I now get an exception for Windows machines when extracting a file to a folder in which some of the files already exist.

This led me to propose #17.

Reverting to jarchivelib or using the change proposed in #17 along with `StandardCopyOption.REPLACE_EXISTING` seems to fix the issue for now.

Here is the stack trace using the latest available version of compress4j:

```java
java.nio.file.FileAlreadyExistsException: C:\Users\username\Documents\fresh-install-1900\folder\file.txt
	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:87)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:236)
	at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:484)
	at java.base/java.nio.file.Files.newOutputStream(Files.java:228)
	at java.base/java.nio.file.Files.copy(Files.java:3160)
	at io.github.compress4j.archivers.IOUtils.copy(IOUtils.java:68)
	at io.github.compress4j.archivers.CommonsArchiveEntry.extract(CommonsArchiveEntry.java:65)
	... <other unrelated parts of the stack>
```

**Comment by austek** [comment]
I'm in the process of re-writing the whole library, and hopefully will incorporate a lot more flexibility into it

**Comment by hhslepicka** [comment]
Addressed via #19
